includes/class-smaily-for-wp.php

```diff
@@ -138,6 +140,15 @@ private function define_public_hooks() {
 	 * @since 3.0.0
 	 */
 	public static function activate() {
+		// Run upgrade scripts before dbDelta, as the latter may delete information.
+		$plugin_version = SMLY4WP_PLUGIN_VERSION;
+		if ( $plugin_version !== get_option( 'smailyforwp_db_version' ) ) {
+			if ( version_compare( $plugin_version, '3.0.0', '=' ) ) {
```

Comment:
The issue I am seeing with this type of comparison is that upgrades could be skipped. For example, if user would be on version 2.3.0 and upgrade to 3.1.0 (or even 3.0.1) they wouldn't get the required upgrade logic that 3.0.0 provides.

Reply:
It should compare the database version, and check if the database is older than 3.0.0